File: kakao/2021/kakao2021_5.py
def t2s(time):
    h, m, s = [int(x) for x in time.split(':')]
    return (h * 60 + m) * 60 + s

def solution(play_time, adv_time, logs):
    pt, at = t2s(play_time), t2s(adv_time)
    d = [0]*360001

    for log in logs:
        st, en = map(t2s, log.split('-'))
        d[st] += 1
        d[en] -= 1

    for i in range(1, 360001):
        d[i] += d[i-1]

    mval, mtime = sum(d[:at]),0
    curval = mval
    for i in range(1, 360001-at):
        curval = curval - d[i-1] +d[i+at-1]
        if curval > mval:
            mval = curval
            mtime = i
    return f'{mtime//3600:02d}:{mtime%3600//60:02d}:{mtime%60:02d}'  

File: kakao/2021/kakao2021_5.java
import java.util.*;
class Solution {
    public String solution(String play_time, String adv_time, String[] logs) {
        int playTime = conv(play_time);
        int advTime = conv(adv_time);
        long playNums[] = new long[playTime+10];

        for(String log : logs){
            String tmp[] = log.split("-");
            playNums[conv(tmp[0])]++;
            playNums[conv(tmp[1])]--;
        }
        for(int i = 0; i < playNums.length-1; i++)
            playNums[i+1] += playNums[i];

        int startTime = 0;
        long timeSum = 0;
        long max = timeSum;
        long maxStartTime = 0;
        while(startTime+advTime <= playTime){
            if(max < timeSum){
                max = timeSum;
                maxStartTime = startTime;
            }
            timeSum -= playNums[startTime];
            timeSum += playNums[startTime+advTime];
            startTime++;
        }
        return String.format("%02d:%02d:%02d",
                maxStartTime/3600,
                (maxStartTime/60)%60,
                maxStartTime%60);
    }

    int conv(String time){
        String splitted[] = time.split(":");
        return Integer.parseInt(splitted[0])*3600
                + Integer.parseInt(splitted[1])*60
                + Integer.parseInt(splitted[2]);
    }
}